allennlp/data/dataset_readers/reading_comprehension/triviaqa.py

```diff
@@ -1,32 +1,280 @@
+
+        else:
+            current_paragraph += " " + _PARAGRAPH_TOKEN + " " + paragraph
+            current_length += paragraph_length
+
+    if current_length > 0:
+        merged_paragraphs.append(current_paragraph)
+
+    # Tokenize the merged paragraphs.
+    merged_paragraph_tokens = [tokenizer.tokenize(paragraph) for paragraph in merged_paragraphs]
+
+    # If ``max_paragraphs`` is None, we want all the merged paragraphs.
+    if max_paragraphs is None:
+        max_paragraphs = len(merged_paragraphs)
+
+    # Get the ranked indexes, from "best" paragraph to worst paragraph.
+    if tfidf_sort:
+        # Sort the paragraphs by their tfidf scores with the question.
+        try:
+            tfidf = TfidfVectorizer(strip_accents="unicode", stop_words="")
+            paragraph_features = tfidf.fit_transform(merged_paragraphs)
+            question_features = tfidf.transform([question])
+            scores = pairwise_distances(question_features, paragraph_features, "cosine").ravel()
+        except ValueError:
+            scores = np.array([0.0] * len(merged_paragraphs))
+
+        # Sort by distance, break ties by original order.
+        ranks: Iterable[int] = [i for i, _ in sorted(enumerate(scores), key=lambda pair: (pair[1], pair[0]))]
+    else:
+        # Keep the paragraphs in their original order.
+        ranks = range(len(merged_paragraphs))
+
+    # Find the indices of paragraphs that have answers.
+    has_answers = [i for i in ranks
+                   if util.find_valid_answer_spans(merged_paragraph_tokens[i], answer_texts)]
+
+    # In the require_answer case, we just return None if no paragraph has the answer.
+    if require_answer and not has_answers:
+        return None
+
+    if require_answer:
+        # Want first_answer to be the first paragraph, and then take the most highly ranked
+        # other topn - 1
+        first_answer = has_answers[0]
+        choices: Iterable[int] = ([first_answer] +
+                                  [i for i in ranks if i != first_answer][:(max_paragraphs - 1)])
+    else:
+        choices = range(min(max_paragraphs, len(merged_paragraphs)))
+
+    merged_texts = [merged_paragraphs[i] for i in choices]
+    merged_tokens = [merged_paragraph_tokens[i] for i in choices]
+    merged_token_spans = [util.find_valid_answer_spans(tokens_i, answer_texts)
+                          for tokens_i in merged_tokens]
+
+    return MergedParagraphs(texts=merged_texts,
+                            tokens=merged_tokens,
+                            token_spans=merged_token_spans,
+                            has_answers=[i for i, choice in enumerate(choices) if choice in has_answers])
+
+
+def process_triviaqa_questions(evidence_path: pathlib.Path,
+                               questions_path: pathlib.Path,
+                               tokenizer: Tokenizer,
+                               max_paragraphs: int = 4,
+                               tfidf_sort: bool = True,
```

Comment:
Presumably anyone using this multi paragraph network will have a tfidf paragraph store sitting in front of the network, they will choose the highest ranked (tfidf similarity) paragraphs for the question and pass them into the network. 
the network then returns them the correct answer. 

taking this into account along with the fact that tfidf_sort is always true, maybe we should just omit it as I don't really see a scenario where you would want to train a system with it set to false. 

What do you guys think?

Reply:
If you want to measure the effect of using TF-IDF to sort versus just taking, e.g., the first N paragraphs, then yes, you want this flag.  This was one of the experiments from the original paper.  AllenNLP is about doing science, so we tend to have options for things that you might experiment with in a paper.